reasoning_gym/algebra/polynomial_multiplication.py

```diff
@@ -143,11 +185,11 @@ def score_answer(self, answer: Optional[str], entry: Dict[str, Any]) -> float:
         metadata = entry["metadata"]
         if answer is not None:
             try:
-                predicted_poly = sp.parse_expr(answer)
-                target_poly = sp.parse_expr(metadata["result"])
+                predicted_poly = sp.Poly(answer)
+                target_poly = sp.Poly(metadata["result"])
 
                 # Check if the difference simplifies to zero (i.e. they are equivalent).
```

Comment:
The implementation seems to accept the representation from the question directly as valid answer:
```
>>> d[11]['question']
'Calculate the following: -25*z**2*(-24*z - 61)'
>>> d.score_answer("-25*z**2*(-24*z - 61)", entry=d[11])
1.0
>>> d[11]['answer']
600*z**3 + 1525*z**2
```
Could we somehow measure how close the submitted answer is to a fully simplified representation?